lib/autokey/configmanager/configmanager.py

```diff
@@ -669,22 +670,41 @@ def check_hotkey_unique(self, modifiers, hotKey, newFilterPattern, targetItem):
         @param newFilterPattern:
         @param targetItem: the phrase for which the hotKey to be used
         """
-        for item in self.allFolders:
-            if model.TriggerMode.HOTKEY in item.modes:
-                if item.modifiers == modifiers and item.hotKey == hotKey and item.filter_matches(newFilterPattern):
-                    return item is targetItem, item
+        item = self.get_item_with_hotkey(modifiers, hotKey, newFilterPattern)
+        if item:
+            return item is targetItem, item
+        else:
+            return True, None
 
-        for item in self.allItems:
-            if model.TriggerMode.HOTKEY in item.modes:
-                if item.modifiers == modifiers and item.hotKey == hotKey and item.filter_matches(newFilterPattern):
-                    return item is targetItem, item
+    def get_item_with_hotkey(self, modifiers, hotKey, newFilterPattern=None):
+        """
+        Gets first item with the specified hotkey. Also checks the
+        special hotkeys configured from the advanced settings dialog.
+        Checks folders first, then phrases, then special hotkeys.
+
+        @param modifiers: modifiers for the hotkey
+        @param hotKey: the hotkey to check
+        @param newFilterPattern:
+        """
+        for item in itertools.chain(self.allFolders, self.allItems):
+            if model.TriggerMode.HOTKEY in item.modes and \
+                    self.item_has_same_hotkey(item,
+                                              modifiers,
+                                              hotKey,
+                                              newFilterPattern):
+                return item
 
         for item in self.globalHotkeys:
             if item.enabled:
-                if item.modifiers == modifiers and item.hotKey == hotKey and item.filter_matches(newFilterPattern):
-                    return item is targetItem, item
+                if self.item_has_same_hotkey(item,
```

Comment:
You can just `and` both conditions here and reduce a nesting depth, like you did in the for loop above (line 690, https://github.com/autokey/autokey/pull/356/files#diff-d8b02a3684c92866db39ed969414c71dR689-R690)